ksql-functional-tests/src/test/resources/query-validation-tests/average-udaf.json

```diff
@@ -0,0 +1,82 @@
+{
+  "comments": [
+    "Tests covering the use of the AVERAGE aggregate function"
+  ],
+  "tests": [
+    {
+      "name": "average int",
+      "statements": [
+        "CREATE STREAM INPUT (VALUE integer) WITH (kafka_topic='test_topic', value_format='JSON');",
+        "CREATE TABLE OUTPUT AS SELECT average(value) AS avg FROM INPUT group by ROWKEY;"
+      ],
+      "inputs": [
+        {"topic": "test_topic", "key": "alice", "value": {"value": 1}},
+        {"topic": "test_topic", "key": "bob", "value": {"value": 2}},
+        {"topic": "test_topic", "key": "alice", "value": {"value": 2}},
+        {"topic": "test_topic", "key": "alice", "value": {"value": 2}},
+        {"topic": "test_topic", "key": "bob", "value": {"value": 2}},
+        {"topic": "test_topic", "key": "bob", "value": {"value": 0}},
+        {"topic": "test_topic", "key": "alice", "value": {"value": null}}
+      ],
+      "outputs": [
+        {"topic": "OUTPUT", "key": "alice", "value": {"AVG": 1.0}},
+        {"topic": "OUTPUT", "key": "bob", "value": {"AVG": 2.0}},
+        {"topic": "OUTPUT", "key": "alice", "value": {"AVG": 1.5}},
+        {"topic": "OUTPUT", "key": "alice", "value": {"AVG": 1.6666666666666667}},
+        {"topic": "OUTPUT", "key": "bob", "value": {"AVG": 2.0}},
+        {"topic": "OUTPUT", "key": "bob", "value": {"AVG": 1.3333333333333333}},
+        {"topic": "OUTPUT", "key": "alice", "value": {"AVG": 1.6666666666666667}}
```

Comment:
Looks like the `null` value is adjusting the average... is this what we want? i.e. do we want the UDF to _ignore_ `null`s, or treat them as zero?

IMHO I think we should ignore them.  Users would them be free to explicitly handle nulls if they wanted e.g.

```sql
SELECT average(IsNull(value, 0.0)) from input group by ROWKEY;
```

Reply:
Null values are ignored. Not sure what you are referring to.

Reply:
Ah, my mistake, I missed the change of key!